azurerm/internal/services/servicebus/validate/servicebus_namespace.go

```diff
@@ -0,0 +1,44 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// validation
+func ServiceBusNamespaceName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+
+	if !regexp.MustCompile("[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}").MatchString(value) {
+		errors = append(errors, fmt.Errorf("%s must be a GUID", k))
+	}
```

Comment:
We should do this instead as the logic as is nothing will pass because you are validating for the name being both a guid and a name
```suggestion
	if regexp.MustCompile("[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}").MatchString(value) {
		return
	}
```